x/evm/precompile/test/export.go

```diff
@@ -308,9 +312,68 @@ func IncrementWasmCounterWithExecuteMulti(
 	s.Require().NoError(err)
 
 	ethTxResp, evmObj, err := deps.EvmKeeper.CallContractWithInput(
-		deps.Ctx, deps.Sender.EthAddr, &precompile.PrecompileAddr_Wasm, true, input,
+		deps.Ctx, deps.Sender.EthAddr, &precompile.PrecompileAddr_Wasm, finalizeTx, input,
 	)
 	s.Require().NoError(err)
 	s.Require().NotEmpty(ethTxResp.Ret)
 	return evmObj
 }
+
+func IncrementWasmCounterWithExecuteMultiViaVMCall(
```

Comment:
Uses *vm.EVM.Call instead of a full tx so that intermediate states can be tested